tuf/utils/x509_test.go

```diff
@@ -239,6 +279,296 @@ func TestECDSAX509PublickeyID(t *testing.T) {
 	require.Equal(t, tufPrivKey.ID(), tufID)
 }
 
+func preserveEnv(name string) func() {
+	if env, has := os.LookupEnv(name); has {
+		os.Unsetenv(name)
+		return func() {
+			os.Setenv(name, env)
+		}
+	}
+
+	return func() {}
+}
+
+func TestExtractPrivateKeyAttributes(t *testing.T) {
+	if os.Getenv(t.Name()) != "1" {
```

Comment:
Its a tricky one. Basically this is a test we try to emulate both environments i.e. one with `GOFIPS=1` and the other when the flag is not set. The workflow is somewhat like this:

- checks if `os.Getenv(t.Name())` is set to 1, initially it would be true, so we defer a function that keeps the value of environment variable in the closure and unsets it. So, rest of the test gets no value `GOFIPS` whether the tests are run with it being set to 1 or not. 
- checks all workflows related to both PKCS#1 and PKCS#8 for no value of `GOFIPS`. 
- spawns a new process that executes this particular test in an emulated environment which has two environment variable, `GOFIPS=1` and `t.Name()=1`.
- checks again if `os.Getenv(t.Name())` is set to 1, this time its false so it does nothing.
- checks all workflows related to both PKCS#1 and PKCS#8 for `GOFIPS=1`.
- spawned process gets done, deferred closure gets called which sets the value of `GOFIPS` to whatever it was before the test.